cylc/flow/tests/host_select.py

```diff
@@ -0,0 +1,186 @@
+"""Test the cylc.flow.host_select module.
+
+NOTE: these are functional tests, for unit tests see the docstrings in
+      the host_select module.
+
+"""
+import socket
+
+import pytest
+
+from cylc.flow.exceptions import HostSelectException
+from cylc.flow.host_select import (
+    select_host,
+    select_suite_host
+)
+from cylc.flow.hostuserutil import get_fqdn_by_host
+from cylc.flow.parsec.exceptions import ListValueError
+from cylc.flow.tests.util import mock_glbl_cfg
+
+
+localhost, localhost_aliases, _ = socket.gethostbyname_ex('localhost')
+localhost_fqdn = get_fqdn_by_host(localhost)
+
+
+def test_hostname_checking():
+    """Check that unknown hosts raise an error"""
+    with pytest.raises(socket.gaierror):
+        select_host(['beefwellington'])
```

Comment:
What's this sudden obsession with beef wellington? Can we have Salmon en croute?

Reply:
If you wish.